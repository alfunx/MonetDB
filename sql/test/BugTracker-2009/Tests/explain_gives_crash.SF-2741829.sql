-- disable parallelism (mitosis & dataflow) to avoid ambiguous results 
set optimizer='sequential_pipe';
\f csv
create table blabla(id integer);
select '#~BeginVariableOutput~#';
explain alter table blabla add constraint dada unique (id);
explain alter table blabla add constraint dada unique (id);
select '#~EndVariableOutput~#';
alter table blabla drop constraint dada;
select '#~BeginVariableOutput~#';
explain alter table blabla add constraint dada unique (id);
select '#~EndVariableOutput~#';
