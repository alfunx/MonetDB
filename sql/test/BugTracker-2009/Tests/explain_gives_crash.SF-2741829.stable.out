stdout of test 'explain_gives_crash.SF-2741829` in directory 'sql/test/BugTracker-2009` itself:


# 22:45:54 >  
# 22:45:54 >   mserver5 "--config=/ufs/niels/scratch/rc/Linux-x86_64/etc/monetdb5.conf" --debug=10 --set gdk_nr_threads=0 --set "monet_mod_path=/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB5:/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB5/lib:/ufs/niels/scratch/rc/Linux-x86_64/lib/MonetDB5/bin" --set "gdk_dbfarm=/ufs/niels/scratch/rc/Linux-x86_64/var/MonetDB5/dbfarm"    --set mapi_open=true --set xrpc_open=true --set mapi_port=35519 --set xrpc_port=48894 --set monet_prompt= --trace  "--dbname=mTests_src_test_BugTracker-2009" --set mal_listing=0 "--dbinit= include sql;" ; echo ; echo Over..
# 22:45:54 >  

# MonetDB server v5.12.0, based on kernel v1.30.0
# Serving database 'mTests_src_test_BugTracker-2009', using 4 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs dynamically linked
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2009 MonetDB B.V., all rights reserved
# Visit http://monetdb.cwi.nl/ for further information
# Listening for connection requests on mapi:monetdb://alf.ins.cwi.nl:35519/
# MonetDB/SQL module v2.30.0 loaded

Ready.
#function user.main():void;
#    clients.quit();
#end main;


# 21:13:40 >  
# 21:13:40 >  mclient -lsql -i --host=alf --port=39638 
# 21:13:40 >  

#~BeginVariableOutput~#
function user.s4_1{autoCommit=true}():void;
"    X_32:void := querylog.define(""explain alter table blabla add constraint dada unique (id);"",""sequential_pipe"",22);"
    X_2 := sql.mvc();
"    sql.catalog(29,""sys"",nil,0);"
"    X_10:bat[:oid,:int] := sql.bind(X_2,""sys"",""blabla"",""id"",0);"
"    X_8:bat[:oid,:oid]  := sql.tid(X_2,""sys"",""blabla"");"
"    X_44 := algebra.subselect(X_10,X_8,nil:int,nil:int,true,true,true);"
"    (X_12,r1_15) := sql.bind(X_2,""sys"",""blabla"",""id"",2);"
"    X_45 := algebra.subselect(r1_15,nil:bat[:oid,:oid],nil:int,nil:int,true,true,true);"
"    X_15:bat[:oid,:int] := sql.bind(X_2,""sys"",""blabla"",""id"",1);"
"    X_47 := algebra.subselect(X_15,X_8,nil:int,nil:int,true,true,true);"
"    X_18 := sql.subdelta(X_44,X_8,X_12,X_45,X_47);"
"    X_19 := sql.projectdelta(X_18,X_10,X_12,r1_15,X_15);"
"    (X_20,r1_26,r2_26) := group.subgroupdone(X_19);"
    X_23 := aggr.count(r2_26);
    X_24:wrd := aggr.sum(r2_26);
    X_25 := calc.isnil(X_24);
"    X_27 := calc.ifthenelse(X_25,0,X_24);"
"    X_28 := calc.!=(X_23,X_27);"
"    sql.assert(X_28,""UPDATE: UNIQUE constraint 'blabla.dada' violated"");"
    sql.exportOperation();
end user.s4_1;
function user.s5_1{autoCommit=true}():void;
"    X_32:void := querylog.define(""explain alter table blabla add constraint dada unique (id);"",""sequential_pipe"",22);"
    X_2 := sql.mvc();
"    sql.catalog(29,""sys"",nil,0);"
"    X_10:bat[:oid,:int] := sql.bind(X_2,""sys"",""blabla"",""id"",0);"
"    X_8:bat[:oid,:oid]  := sql.tid(X_2,""sys"",""blabla"");"
"    X_44 := algebra.subselect(X_10,X_8,nil:int,nil:int,true,true,true);"
"    (X_12,r1_15) := sql.bind(X_2,""sys"",""blabla"",""id"",2);"
"    X_45 := algebra.subselect(r1_15,nil:bat[:oid,:oid],nil:int,nil:int,true,true,true);"
"    X_15:bat[:oid,:int] := sql.bind(X_2,""sys"",""blabla"",""id"",1);"
"    X_47 := algebra.subselect(X_15,X_8,nil:int,nil:int,true,true,true);"
"    X_18 := sql.subdelta(X_44,X_8,X_12,X_45,X_47);"
"    X_19 := sql.projectdelta(X_18,X_10,X_12,r1_15,X_15);"
"    (X_20,r1_26,r2_26) := group.subgroupdone(X_19);"
    X_23 := aggr.count(r2_26);
    X_24:wrd := aggr.sum(r2_26);
    X_25 := calc.isnil(X_24);
"    X_27 := calc.ifthenelse(X_25,0,X_24);"
"    X_28 := calc.!=(X_23,X_27);"
"    sql.assert(X_28,""UPDATE: UNIQUE constraint 'blabla.dada' violated"");"
    sql.exportOperation();
end user.s5_1;
#~EndVariableOutput~#
#~BeginVariableOutput~#
function user.s9_1{autoCommit=true}():void;
"    X_32:void := querylog.define(""explain alter table blabla add constraint dada unique (id);"",""sequential_pipe"",22);"
    X_2 := sql.mvc();
"    sql.catalog(29,""sys"",nil,0);"
"    X_10:bat[:oid,:int] := sql.bind(X_2,""sys"",""blabla"",""id"",0);"
"    X_8:bat[:oid,:oid]  := sql.tid(X_2,""sys"",""blabla"");"
"    X_44 := algebra.subselect(X_10,X_8,nil:int,nil:int,true,true,true);"
"    (X_12,r1_15) := sql.bind(X_2,""sys"",""blabla"",""id"",2);"
"    X_45 := algebra.subselect(r1_15,nil:bat[:oid,:oid],nil:int,nil:int,true,true,true);"
"    X_15:bat[:oid,:int] := sql.bind(X_2,""sys"",""blabla"",""id"",1);"
"    X_47 := algebra.subselect(X_15,X_8,nil:int,nil:int,true,true,true);"
"    X_18 := sql.subdelta(X_44,X_8,X_12,X_45,X_47);"
"    X_19 := sql.projectdelta(X_18,X_10,X_12,r1_15,X_15);"
"    (X_20,r1_26,r2_26) := group.subgroupdone(X_19);"
    X_23 := aggr.count(r2_26);
    X_24:wrd := aggr.sum(r2_26);
    X_25 := calc.isnil(X_24);
"    X_27 := calc.ifthenelse(X_25,0,X_24);"
"    X_28 := calc.!=(X_23,X_27);"
"    sql.assert(X_28,""UPDATE: UNIQUE constraint 'blabla.dada' violated"");"
    sql.exportOperation();
end user.s9_1;
#~EndVariableOutput~#

# 11:37:28 >  
# 11:37:28 >  Done.
# 11:37:28 >  

